controllers/hcpvaultsecretsapp_controller_test.go

```diff
@@ -1224,3 +1228,130 @@ func Test_makeShadowObjKey(t *testing.T) {
 		})
 	}
 }
+
+func Test_CleanupOrphanedShadowSecrets(t *testing.T) {
+	deletionTimestamp := metav1.Now()
+
+	hvsApp := secretsv1beta1.HCPVaultSecretsApp{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       HCPVaultSecretsApp.String(),
+			APIVersion: secretsv1beta1.GroupVersion.Version,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:  "hvs-app-ns",
+			Name:       "hvs-app",
+			Finalizers: []string{hcpVaultSecretsAppFinalizer},
+		},
+	}
+
+	tests := map[string]struct {
+		o                                    *secretsv1beta1.HCPVaultSecretsApp
+		secret                               *corev1.Secret
+		isHCPVaultSecretsAppDeletionExpected bool
+		isShadowSecretDeletionExpected       bool
+	}{
+		"HCPVaultSecretsApp and shadow secret marked for deletion, HCPVaultSecretsApp is owner of shadow secret": {
+			o: hvsApp.DeepCopy(),
+			secret: &corev1.Secret{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       VaultDynamicSecret.String(),
+					APIVersion: secretsv1beta1.GroupVersion.Version,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace:         common.OperatorNamespace,
+					Name:              "shadow-secret",
+					DeletionTimestamp: &deletionTimestamp,
+					Finalizers:        []string{vaultDynamicSecretFinalizer},
+					Labels: map[string]string{
+						hvsaLabelPrefix + "/name":      hvsApp.GetName(),
+						hvsaLabelPrefix + "/namespace": hvsApp.GetNamespace(),
+						"app.kubernetes.io/component":  "hvs-dynamic-secret-cache",
+						labelOwnerRefUID:               string(hvsApp.GetUID()),
+					},
+				},
+			},
+			isHCPVaultSecretsAppDeletionExpected: true,
+			isShadowSecretDeletionExpected:       true,
+		},
+		"HCPVaultSecretsApp not owner of shadow secret, shadow secret marked for deletion": {
+			o: hvsApp.DeepCopy(),
+			secret: &corev1.Secret{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       VaultDynamicSecret.String(),
+					APIVersion: secretsv1beta1.GroupVersion.Version,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace:         common.OperatorNamespace,
+					Name:              "shadow-secret",
+					DeletionTimestamp: &deletionTimestamp,
+					Finalizers:        []string{vaultDynamicSecretFinalizer},
+					Labels: map[string]string{
+						"app.kubernetes.io/component": "hvs-dynamic-secret-cache",
```

Comment:
Might be good to have a case with a Secret in the operator namespace that doesn't have this label, and verify that it's still around after the cleanup is run.